hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/storage/HoodieHFileWriter.java

```diff
@@ -121,17 +121,10 @@ public void writeAvro(String recordKey, IndexedRecord object) throws IOException
 
     if (hfileConfig.useBloomFilter()) {
       hfileConfig.getBloomFilter().add(recordKey);
-      if (minRecordKey != null) {
-        minRecordKey = minRecordKey.compareTo(recordKey) <= 0 ? minRecordKey : recordKey;
-      } else {
+      if (minRecordKey == null) {
```

Comment:
this is effectively not computing the min/max, right? Not sure if we should do this change.

Reply:
We are effectively dependent on the higher layer sorting in a much lower layer. We should preserve the min/max computation here IMO

Reply:
Hi @vinothchandar, Thanks for reviewing. it not computing the min/max. it only use the first recordKey and the last recordKey as min/max(HoodieSortedMergeHandle/BaseSparkCommitActionExecutor already compare the input records in order by recordKey) . It's like hbase store keyRange(firstKey/lastKey) as 
[https://github.com/apache/hbase/blob/rel/1.2.3/hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileWriterV2.java#L292](https://github.com/apache/hbase/blob/rel/1.2.3/hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileWriterV2.java#L292)
and actually we also can get min/max RecordKey from HFile native thought `HFileReaderV3#getFirstKey()` and `HFileReaderV3#getLastRowKey()` from load-on-open section

and i think we can use current implement(put min/max in FileInfo map). maybe we will add more properties. for example: add recordCount so we can  choose seekto or loadall in `HoodieHFileReader#filterRowKeys`

Reply:
I feel this should be fine. We ought to sort entries before writing to Hfile anyways. So, if some callers is gonna try writing unsorted entries, hfile write will fail anyways.

Reply:
@vinothchandar : please do respond when you can. if the arguments looks convincing?

Reply:
@yui2010 : I synced up with Vinoth on this. He is fine with your optimization. If there are any pending comments to be addressed, can you please address and let me know once the patch is ready to be reviewed again.

Reply:
@nsivabalan thinks for your review ，this change will be more effectively to compute min/max IMO . and  we can discuss more if I missed some logic.